### PR TITLE
Ptcd 177 fa

### PR DESCRIPTION
**Pull request by MrKevJoy**
- Initial work required to sync online courses to azure search index.

### CODE REVIEW

File: src/Dfc.ProviderPortal.ChangeFeedListener/Helpers/FaocSearchServiceWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        public async Task<IEnumerable<IndexingResult>> UploadFaocBatch(IEnumerable<FaocEntry> documents)
+        {
+            var batch = IndexBatch.MergeOrUpload(documents);
+            var indexResult = await _adminIndex.Documents.IndexAsync(batch);
+
+            return indexResult.Results;
+        }
+
+        public async Task<DocumentIndexResult> DeleteStaleDocuments(IEnumerable<FaocEntry> documents)
```
Comment (gunndabad) [inline src/Dfc.ProviderPortal.ChangeFeedListener/Helpers/FaocSearchServiceWrapper.cs:39]: Don't need to return anything here

File: src/Dfc.ProviderPortal.ChangeFeedListener/Settings/FaocSearchServiceSettings.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using Dfc.ProviderPortal.ChangeFeedListener.Interfaces;
+
+namespace Dfc.ProviderPortal.ChangeFeedListener.Settings
+{
+    public class FaocSearchServiceSettings : IFaocSearchServiceSettings
+    {
+        public string SearchService { get; set; }
```
Comment (gunndabad) [inline src/Dfc.ProviderPortal.ChangeFeedListener/Settings/FaocSearchServiceSettings.cs:7]: Not sure we need most of this - key is probably sufficient

File: src/Dfc.ProviderPortal.ChangeFeedListener/Settings/FaocSearchServiceSettings.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        public string ApiUrl { get; set; }
+        public string ApiVersion { get; set; }
+        public string QueryKey { get; set; }
+        public string AdminKey { get; set; }
+        public string Index { get; set; }
+        public int DefaultTop { get; set; }
+        public decimal? RegionSearchBoost { get; set; }
+        public decimal? SubRegionSearchBoost { get; set; }
```
Comment (soharadesign) [inline src/Dfc.ProviderPortal.ChangeFeedListener/Settings/FaocSearchServiceSettings.cs:15]: question: do we need to remove these? We're not doing any region/subregion searching for FAOC

Comment (MrKevJoy) [reply src/Dfc.ProviderPortal.ChangeFeedListener/Settings/FaocSearchServiceSettings.cs]: removed these now.

File: src/Dfc.ProviderPortal.ChangeFeedListener/Settings/FaocSearchServiceSettings.cs (hunk 2 above)
Comment (MrKevJoy) [reply src/Dfc.ProviderPortal.ChangeFeedListener/Settings/FaocSearchServiceSettings.cs:7]: removed most of them now, kept the 4 being used.

File: src/Dfc.ProviderPortal.ChangeFeedListener/Helpers/FaocSearchServiceWrapper.cs (hunk 1 above)
Comment (MrKevJoy) [reply src/Dfc.ProviderPortal.ChangeFeedListener/Helpers/FaocSearchServiceWrapper.cs]: removed

**Pull request merged**